src/elements/FunctionsPanelCode/FunctionsPanelCodeView.js

```diff
@@ -60,51 +67,71 @@ const FunctionsPanelCodeView = ({
-            }
-            onBlur={() => setNewFunctionBuildImage(data.build_image)}
-            type="text"
-            value={data.build_image}
-            wrapperClassName="build-image"
-          />
-          <Input
-            disabled={data.image.length > 0}
-            floatingLabel
-            label="Base image"
-            onChange={base_image =>
-              setData(state => ({ ...state, base_image }))
-            }
-            onBlur={() => setNewFunctionBaseImage(data.base_image)}
-            type="text"
-            value={data.base_image}
-            wrapperClassName="base-image"
-          />
+          <div className="code__handler">
+            <Input
+              floatingLabel
+              label="Handler"
+              onChange={handleHandlerChange}
+              onBlur={handleHandlerOnBlur}
+              required={!isHandlerValid}
+              requiredText="This field is required"
+              type="text"
+              value={data.handler}
+              wrapperClassName="handler"
+            />
+          </div>
+          <div className="code__existing-image">
+            <RadioButtons
+              onChangeCallback={setRadioButtonsValue}
+              elements={codeOptions}
+              className="radio-buttons__block"
+              selectedValue={radioButtonsValue}
+            />
+            <div className="code__images-inputs">
+              <Input
+                className="input__wide"
+                disabled={radioButtonsValue !== 'existingImage'}
+                floatingLabel
+                label="Image name"
+                onChange={image => setData(state => ({ ...state, image }))}
+                onBlur={() => setNewFunctionImage(data.image)}
+                type="text"
+                value={data.image}
+                wrapperClassName="image-name"
+                tip="The name of the function‘s container image"
+              />
+              <div>
+                <Input
+                  className="input__wide"
+                  disabled={radioButtonsValue !== 'newImage'}
+                  floatingLabel
+                  label="Resulting Image"
+                  onChange={build_image =>
+                    setData(state => ({ ...state, build_image }))
+                  }
+                  onBlur={() => setNewFunctionBuildImage(data.build_image)}
+                  type="text"
+                  value={data.build_image}
+                  wrapperClassName="build-image"
+                  tip="The name of the built container image"
+                />
+                <Input
+                  className="input__wide"
+                  disabled={radioButtonsValue !== 'newImage'}
+                  floatingLabel
+                  label="Base image"
+                  onChange={base_image =>
+                    setData(state => ({ ...state, base_image }))
+                  }
+                  onBlur={() => setNewFunctionBaseImage(data.base_image)}
```

Comment:
wee need to add check so it doesn't fire every time even if we didn't changed anything